Source/Core/Core/DSP/DSPCore.cpp

```diff
@@ -75,22 +78,13 @@ static bool VerifyRoms(const SDSP& dsp)
       return false;
   }
 
-  if (rom_idx == 1)
-  {
-    Host::OSD_AddMessage("You are using an old free DSP ROM made by the Dolphin Team.", 6000);
-    Host::OSD_AddMessage("Only games using the Zelda UCode will work correctly.", 6000);
-  }
-  else if (rom_idx == 2 || rom_idx == 3)
-  {
-    Host::OSD_AddMessage("You are using a free DSP ROM made by the Dolphin Team.", 8000);
-    Host::OSD_AddMessage("All Wii games will work correctly, and most GameCube games", 8000);
-    Host::OSD_AddMessage("should also work fine, but the GBA/CARD UCodes will not work.", 8000);
-  }
-  else if (rom_idx == 4)
+  if (rom_idx >= 1 && rom_idx <= 5)
   {
-    Host::OSD_AddMessage("You are using a free DSP ROM made by the Dolphin Team.", 8000);
-    Host::OSD_AddMessage("All Wii games will work correctly, and most GameCube games", 8000);
-    Host::OSD_AddMessage("should also work fine, but the CARD UCode will not work.", 8000);
+    if (AskYesNoFmtT("You are using an old free DSP ROM made by the Dolphin Team.\n"
+                     "Due to emulation inaccuracy improvements, this ROM longer works correctly.\n"
+                     "Would you like to stop now to fix the problem?\n"
+                     "If you select \"No\", audio might be garbled."))
+      return false;
```

Comment:
Do we want to include some info on _what_ the user could do to fix the problem (for example, delete/replace the file?)

Also, since the `if` condition spans over multiple lines, our code style wants curlies here for readability.

Reply:
Done.  I've also changed "emulation inaccuracy improvements" to "emulation accuracy improvements".

Reply:
"emulation inaccuracy improvements" sounds very strange, I'd drop the "in-" prefix.